executor/read/pattern_executor.rs

```diff
@@ -139,11 +232,11 @@ impl PatternExecutor {
                         unreachable!();
                     };
                     let branch = &mut disjunction.branches[branch_index.0];
-                    let suspend_point_len_before = suspend_point_accumulator.len();
+                    let suspend_tracker_state_before = suspend_point_context.record_nested_pattern_entry();
```

Comment:
If this type becomes `SuspendPointCount` then this is `suspend_point_count_before`, which then kinda makes sense to compare it to the count of suspend points after and see if somethiing needs to be done?

Reply:
Yeah I was wondering whether to add the depth as well then decided not to.